ocs_ci/deployment/terraform.py

```diff
@@ -0,0 +1,50 @@
+"""
+This module contains terraform specific methods and classes needed
+for deployment on vSphere platform
+"""
+import logging
+
+from ocs_ci.utility.utils import run_cmd
+
+logger = logging.getLogger(__name__)
+
+
+class Terraform(object):
+    """
+    Wrapper for terraform
+    """
+    def __init__(self, path):
+        self.path = path
+
+    def initialize(self):
+        """
+        Initialize a working directory containing Terraform configuration files
+        """
+        logger.info("Initializing terraform work directory")
+        run_cmd(f"terraform init {self.path}")
+
+    def apply(self, tfvars, bootstrap_complete=False):
+        """
+        Apply the changes required to reach the desired state of the configuration
+
+        Args:
+            tfvars (str): path to terraform.tfvars file
+            bootstrap_complete (bool): Removes bootstrap node if True
+
+        """
+        if bootstrap_complete:
+            cmd = f"terraform apply '-var-file={tfvars}' -auto-approve -var bootstrap_complete=true '{self.path}'"
+        else:
+            cmd = f"terraform apply '-var-file={tfvars}' -auto-approve '{self.path}'"
+        run_cmd(cmd)
+
+    def destroy(self, tfvars):
+        """
+        Destroys the cluster
+
+        Args:
+            tfvars (str): path to terraform.tfvars file
+
+        """
+        logger.info("Destroying the cluster")
+        run_cmd(f"terraform destroy '-var-file={tfvars}' -auto-approve {self.path}")
```

Comment:
On the jenkins we are hitting this issue when I tried to run on already deployed cluster (unsuccessfully):
 ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: terraform destroy '-var-file=/home/jenkins/current-cluster-dir/openshift-cluster-dir/terraform.tfvars' -auto-approve /home/jenkins/workspace/qe-destroy-ocs-cluster/ocs-ci/external/installer/upi/vsphere/.
E           Error is Failed to load Terraform configuration or plan: open /home/jenkins/workspace/qe-destroy-ocs-cluster/ocs-ci/external/installer/upi/vsphere/: no such file or directory

I think that it count's that you have the installer/upi/vsphere/ in external folder but as jenkins cleaning up the repository between executions you will not have it there and it will have to be cloned again I think.

Reply:
Its looking for the installer folder which was cleaned by jenkins.
As per Discussion, changing the current directory when running terraform that suits for jenkins job

update code in commit 2f4387e6dd6f3fe228c27a1f9f696217963f660c